reana/cli.py

```diff
@@ -877,8 +883,12 @@ def git_push(full, component):  # noqa: D301
 @click.option('--build-arg', '-b', default='', multiple=True,
               help='Any build arguments? (e.g. `-b DEBUG=true`)')
 @click.option('--no-cache', is_flag=True)
+@click.option('--dev-tag', is_flag=True,
```

Comment:
Hmm, `-t` has the default value `latest`, what about using a special value like `-t dirty` that would generate proper dirty tag `v0.4.0-9-gd68e970`? It may be more easy than having an extra forcing flag `--dev-tag` since people can misunderstand what would be the result of various combinations of `-t foo --dev-tag`.  If we go for a sinle option `-t`, we could simply explain in the docstring all the various options that `-t` understands, a bit like `-c CLUSTER` and `-c .` and `-c ALL` which people may already know so that could "feel at home" when they see similar behaviour in other options too.

Reply:
:+1: and perhaps we can consider using dynamic `git describe ---dirty` value as default, unless `-t` is used in which case static `-t` value would take over?  (This would mean we'd have to use `reana-dev docker-build -t latest` locally when developing, but this is easy.)  The dynamic default value would probably make more sense than `latest`...